{"Modified":"2021-07-27T17:25:00","Published":"2021-05-14T20:15:00","access":{"authentication":"NONE","complexity":"LOW","vector":"LOCAL"},"assigner":"security-advisories@github.com","capec":[],"cvss":2.1,"cvss-time":"2021-07-27T17:25:00","cvss-vector":"AV:L/AC:L/Au:N/C:N/I:N/A:P","cwe":"CWE-754","id":"CVE-2021-29544","impact":{"availability":"PARTIAL","confidentiality":"NONE","integrity":"NONE"},"last-modified":"2021-07-27T17:25:00","references":["https://github.com/tensorflow/tensorflow/commit/20431e9044cf2ad3c0323c34888b192f3289af6b","https://github.com/tensorflow/tensorflow/security/advisories/GHSA-6g85-3hm8-83f9"],"summary":"TensorFlow is an end-to-end open source platform for machine learning. An attacker can trigger a denial of service via a `CHECK`-fail in `tf.raw_ops.QuantizeAndDequantizeV4Grad`. This is because the implementation(https://github.com/tensorflow/tensorflow/blob/95078c145b5a7a43ee046144005f733092756ab5/tensorflow/core/kernels/quantize_and_dequantize_op.cc#L162-L163) does not validate the rank of the `input_*` tensors. In turn, this results in the tensors being passes as they are to `QuantizeAndDequantizePerChannelGradientImpl`(https://github.com/tensorflow/tensorflow/blob/95078c145b5a7a43ee046144005f733092756ab5/tensorflow/core/kernels/quantize_and_dequantize_op.h#L295-L306). However, the `vec<T>` method, requires the rank to 1 and triggers a `CHECK` failure otherwise. The fix will be included in TensorFlow 2.5.0. We will also cherrypick this commit on TensorFlow 2.4.2 as this is the only other affected version.","vulnerable_configuration":[{"id":"cpe:2.3:a:google:tensorflow:2.4.0:-:*:*:*:*:*:*","title":"cpe:2.3:a:google:tensorflow:2.4.0:-:*:*:*:*:*:*"},{"id":"cpe:2.3:a:google:tensorflow:2.4.0:rc0:*:*:*:*:*:*","title":"cpe:2.3:a:google:tensorflow:2.4.0:rc0:*:*:*:*:*:*"},{"id":"cpe:2.3:a:google:tensorflow:2.4.0:rc1:*:*:*:*:*:*","title":"cpe:2.3:a:google:tensorflow:2.4.0:rc1:*:*:*:*:*:*"},{"id":"cpe:2.3:a:google:tensorflow:2.4.0:rc2:*:*:*:*:*:*","title":"cpe:2.3:a:google:tensorflow:2.4.0:rc2:*:*:*:*:*:*"},{"id":"cpe:2.3:a:google:tensorflow:2.4.0:rc3:*:*:*:*:*:*","title":"cpe:2.3:a:google:tensorflow:2.4.0:rc3:*:*:*:*:*:*"},{"id":"cpe:2.3:a:google:tensorflow:2.4.0:rc4:*:*:*:*:*:*","title":"cpe:2.3:a:google:tensorflow:2.4.0:rc4:*:*:*:*:*:*"},{"id":"cpe:2.3:a:google:tensorflow:2.4.1:*:*:*:*:*:*:*","title":"cpe:2.3:a:google:tensorflow:2.4.1:*:*:*:*:*:*:*"}],"vulnerable_configuration_cpe_2_2":[],"vulnerable_product":["cpe:2.3:a:google:tensorflow:2.4.0:-:*:*:*:*:*:*","cpe:2.3:a:google:tensorflow:2.4.0:rc0:*:*:*:*:*:*","cpe:2.3:a:google:tensorflow:2.4.0:rc1:*:*:*:*:*:*","cpe:2.3:a:google:tensorflow:2.4.0:rc2:*:*:*:*:*:*","cpe:2.3:a:google:tensorflow:2.4.0:rc3:*:*:*:*:*:*","cpe:2.3:a:google:tensorflow:2.4.0:rc4:*:*:*:*:*:*","cpe:2.3:a:google:tensorflow:2.4.1:*:*:*:*:*:*:*"]}